kapitan/utils.py

```diff
@@ -212,6 +212,54 @@ def increase_indent(self, flow=False, indentless=False):
         return super(PrettyDumper, self).increase_indent(flow, False)
 
 
+def multiline_str_presenter(dumper, data):
+    """
+    Configures yaml for dumping multiline strings with given style.
+    By default, strings are getting dumped with style='"'.
+    Ref: https://github.com/yaml/pyyaml/issues/240#issuecomment-1018712495
+    """
+    # get parsed args from cached.py
+    try:
+        cached_dict = cached.as_dict()
+        args = vars(cached_dict["args"]["compile"])
+
+        style = args["multiline_string_style"]
+    except KeyError:
+        style = None
```

Comment:
you should be able to use cached instead like this:

```
compile_args = cached.args
style = args.get("multiline_string_style", None)
```

Reply:
Hi @ramaro,
I don't understand how the args in cached.py are stored and especially why so.
Your suggestion doesn't work, because the `compile_args`-variable is just a dictionary with a single `compile`-key. So the wanted arg-key cannot get found and the default-value gets taken. 
Am I doing something wrong or not seeing anything?

If we print the `compile_args`-variable and the received value for style we get:
Running with `$ kapitan compile --multiline-string-style literal`
```
{'compile': Namespace(subparser_name='compile', [...], multiline_string_style='literal', [...], name='compile')}
Received value for style: None
```
The mentioned `compile`-key has a Namespace object as its value. You can resolve this object by using `vars()` to get the actual dictionary with the args and their values.

I would suggest something like this: 
```python
compile_args = vars(cached.args.get("compile"))
style = compile_args.get("multiline_string_style", None)
```  
This works fine and is a solution without using `try-catches`.  
Again, please let me know, if I'm doing something not as intended.

Reply:
Hmm sorry, my bad for commenting late at night. I meant to suggest that the following instead will grab either the default or flag value in argparse without any conversions

```
style = cached.args["compile"].multiline_string_style
if style bla:
  foobar
```

Reply:
**Update:**
The unittest `test_cli_ref_reveal_recursive_dir` in `test_cli.py` fails with my suggestion. I had this issue before and used `try-except` to make this test not fail.

I know, that if a test fails, we shouldn't wrap everything in a `try-except`-block to make it work.

Maybe the test isn't working correctly, because it throws
```
File "/home/matteo/kapitan_nexenio/kapitan/utils.py", line 222, in multiline_str_presenter
    compile_args = vars(cached.args.get("compile"))
TypeError: vars() argument must have __dict__ attribute
```
This is weird, because in manual use the error doesn't come up.

What are your ideas on that?

Reply:
Ah yes, good point. This is not a big deal though! Both your code and the test are working as expected.
The `test_cli_ref_reveal_recursive_dir` is testing the cli, therefore command line arguments are hardcoded within the test via sys.argv.

So I think the safest way is to adapt your code to check if `compile` is passed (which is always then when not running kapitan from a test). Something like this should work:

```py
compile = cached.args.get("compile", None)
style = None
if compile:
  style = cached.args["compile"].multiline_string_style
if style bla:
  foobar
```

Reply:
This is a good idea. Only change: I renamed `compile` to `compile_args`.

Reply:
thanks!